src/SharpLearning.Backend.TensorFlow.Test/TensorFlowRawMnistTest.cs

```diff
@@ -0,0 +1,344 @@
+
+                // Prediction: Get min distance index (Nearest neighbor)
+                TFOutput pred = g.ArgMin(distance, g.Const(0));
+
+                var accuracy = 0f;
+                // Loop over the test data
+                for (int i = 0; i < testCount; i++)
+                {
+                    var runner = s.GetRunner();
+
+                    // Get nearest neighbor
+
+                    var result = runner.Fetch(pred).Fetch(distance).AddInput(trainingInput, trainingImages).AddInput(xte, Extract(testImages, i)).Run();
+                    var r = result[0].GetValue();
+                    var tr = result[1].GetValue();
+                    var nn_index = (int)(long)result[0].GetValue();
+
+                    // Get nearest neighbor class label and compare it to its true label
+                    //Console.WriteLine($"Test {i}: Prediction: {ArgMax(trainingLabels, nn_index)} True class: {ArgMax(testLabels, i)} (nn_index={nn_index})");
+                    if (ArgMax(trainingLabels, nn_index) == ArgMax(testLabels, i))
+                        accuracy += 1f / testImages.Length;
+                }
+                Console.WriteLine("Accuracy: " + accuracy);
+                Trace.WriteLine("Accuracy: " + accuracy);
+            }
+        }
+        
+        int ArgMax(float[,] array, int idx)
+        {
+            float max = -1;
+            int maxIdx = -1;
+            var l = array.GetLength(1);
+            for (int i = 0; i < l; i++)
+                if (array[idx, i] > max)
+                {
+                    maxIdx = i;
+                    max = array[idx, i];
+                }
+            return maxIdx;
+        }
+
+        public float[] Extract(float[,] array, int index)
+        {
+            var n = array.GetLength(1);
+            var ret = new float[n];
+
+            for (int i = 0; i < n; i++)
+                ret[i] = array[index, i];
+            return ret;
+        }
+    }
+
+    // Below copied from: https://github.com/migueldeicaza/TensorFlowSharp/blob/master/Learn/Datasets/MNIST.cs
+    // Do NOT use this code as foundation for other things, quality is not great!
+
+    // Stores the per-image MNIST information we loaded from disk 
+    //
+    // We store the data in two formats, byte array (as it came in from disk), and float array
+    // where each 0..255 value has been mapped to 0.0f..1.0f
+    public struct MnistImage
+    {
+        public int Cols, Rows;
+        public byte[] Data;
+        public float[] DataFloat;
+
+        public MnistImage(int cols, int rows, byte[] data)
+        {
+            Cols = cols;
+            Rows = rows;
+            Data = data;
+            DataFloat = new float[data.Length];
+            for (int i = 0; i < data.Length; i++)
+            {
+                DataFloat[i] = Data[i] / 255f;
+            }
+        }
+    }
+
+    // Helper class used to load and work with the Mnist data set
+    public class Mnist
```

Comment:
@nietras This might not be the ideal way to represent and load the mnist dataset. But I guess it will suffice for initial testing.

Reply:
@mdabros nah it is pretty dirty and quick way of loading as I said.